src/main/java/com/denizenscript/ddiscordbot/DiscordCommand.java

```diff
@@ -171,6 +213,45 @@ public void execute(ScriptEntry scriptEntry) throws CommandExecutionException {
                 }
                 dDiscordBot.instance.connections.get(id.asString()).client.getChannelByID(channel.asLong()).sendMessage(message.asString());
                 break;
+            case ROLE:
+                if (user == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no user given!");
+                    return;
+                }
+                if (guild == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no guild given!");
+                    return;
+                }
+                if (role == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no role given!");
+                    return;
+                }
+                if (message == null) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: no action given!");
+                    return;
+                }
+                if (!dDiscordBot.instance.connections.containsKey(id.asString())) {
+                    dB.echoError(scriptEntry.getResidingQueue(), "Failed to role: unknown ID!");
+                    return;
+                }
+                IDiscordClient client_d = dDiscordBot.instance.connections.get(id.asString()).client;
+                IGuild guild_d = client_d.getGuildByID(guild.asLong());
+                switch (message.toString().toLowerCase()) {
+                    case "remove":
+                        client_d.getUserByID(user.asLong()).removeRole(guild_d.getRoleByID(role.asLong()));
+                        break;
+                    case "add":
+                        client_d.getUserByID(user.asLong()).addRole(guild_d.getRoleByID(role.asLong()));
+                        break;
+                    case "set":
+                        guild_d.editUserRoles(client_d.getUserByID(user.asLong()), new IRole[]{guild_d.getRoleByID(role.asLong())});

```

Comment:
No.

Reply:
What's wrong about setting a user's role? It's more or less to make sure that the user only has that role and nothing else. (Unless we had tags to check)

Reply:
Add tags to check.
That's poor server operation in general, as far as Discord, to delete all roles with disregard...